src/overlays/actors/ovl_Obj_Bombiwa/z_obj_bombiwa.c

```diff
@@ -22,18 +30,128 @@ const ActorInit Obj_Bombiwa_InitVars =
     (ActorFunc)ObjBombiwa_Update,
     (ActorFunc)ObjBombiwa_Draw,
 };
-*/
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Bombiwa/func_80B915F0.s")
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Bombiwa/ObjBombiwa_Init.s")
+static ColliderCylinderInit colliderInit =
+{
+    0x0C, 0x00, 0x0D, 0x39, 
+    0x20, 0x01, 0x00, 0x00, 
+    0x00, 0x00, 0x00, 0x00, 
+    0x00000000, 0x00, 0x00, 
+    0x00, 0x00, 0x4FC1FFFE, 
+    0x00, 0x00, 0x00, 0x00, 
+    0x00, 0x01, 0x01, 0x00,
+    0x0037, 0x0046, 0x0000, 
+    0x0000, 0x0000, 0x0000
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Bombiwa/ObjBombiwa_Destroy.s")
+static u32 damageChart[] = 
+{
+    0x0000000C, 0x003CFF00
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Bombiwa/func_80B91738.s")
+static InitChainEntry initChain[] = 
+{
+    ICHAIN_VEC3F_DIV1000(scale, 0x64, ICHAIN_CONTINUE),
+    ICHAIN_F32(unk_F4, 0x7D0, ICHAIN_CONTINUE),
+    ICHAIN_F32(unk_F8, 0x15E, ICHAIN_CONTINUE),
+    ICHAIN_F32(unk_FC, 0x3E8, ICHAIN_STOP)
+};
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_Obj_Bombiwa/ObjBombiwa_Update.s")
+static s16 effectScales[] = 
+{
+    17, 14, 10, 8, 7, 5, 3, 2
+};
+
+extern Gfx* D_060009E0; //dlist
+
+static void ObjBombiwa_InitCollision(ObjBombiwa* this, GlobalContext* globalCtx)
+{
+    ObjBombiwa* thisLocal = this;
+    ActorCollider_AllocCylinder(globalCtx, &thisLocal->collider);
+    ActorCollider_InitCylinder(globalCtx, &thisLocal->collider, &thisLocal->actor, &colliderInit);
+    ActorCollider_Cylinder_Update(&thisLocal->actor, &thisLocal->collider);
+}
+
+static void ObjBombiwa_Init(ObjBombiwa* this, GlobalContext* globalCtx)
+{
+    Actor_ProcessInitChain(&this->actor, initChain);
+    ObjBombiwa_InitCollision(this, globalCtx);
+    if ((Flags_GetSwitch(globalCtx, this->actor.params & 0x3F) != 0))
+    {
+        Actor_Kill(&this->actor);
+        return;
```

Comment:
I believe you could get rid of this return and make the block below and else of the outer if

Reply:
I don't really agree with removing this return since the blocks have such wildly different sizes. I think it looks cleaner this way.

Reply:
I think its better in the sense that you can scan the code and quickly realize that it is an if else. Personally dont feel like the aesthetics of the code block sizes should matter for that